modules/serverless-creating-subscriptions.adoc

```diff
@@ -56,5 +59,19 @@ spec:
 . Apply the YAML file by entering:
 +
 ----
-$ oc apply -f <FILENAME>
+$ oc apply -f <filename>
+----
+
+== Create a subscription by using the `kn` CLI
+
+.Procedure
+
+. Create a Subscription object to connect a channel to a service, by entering the following command:
++
+
+[source,terminal]
+----
+$ kn subscription create <subscription-name> --channel <channel-name> --mysink <type:sink-name>
```

Comment:
Adding the command signature and available options first and then specific examples:

Command signature:
```
kn subscription create <subscription_name> --channel <G:V:K>:<channel_name> --sink <sink_prefix>:<sink_name> --sink-reply <sink_prefix>:<sink_name> --sink-dead-letter <sink_prefix>:<sink_name>
```

- `--channel` flag: 

This is required flag while creating the subscription. The `<G:V:K>` part is optional and of format `Group:Version:Kind` of the channel to subscribe to, you can use explicit approach and specify exact GVK for the channel and channel name.
For example: `--channel messaging.knative.dev:v1alpha1:KafkaChannel:mychannel`

You can also omit the GVK completely and only specify the channel name, for example `--channel mychannel`.

Note: `--channel` flag is only available while creating the subscription and thereafter its immutable, i.e. you can not update channel of the subscription using `kn subscription update`.


- `--sink` flag:

Addressable sink for events. You can specify a broker, Knative service or URI. Examples: '--sink broker:nest' for a broker 'nest', '--sink https://event.receiver.uri' for an URI with an 'http://' or 'https://' schema, '--sink ksvc:receiver' or simply '--sink receiver' for a Knative service 'receiver'. If a prefix is not provided, it is considered as a Knative service.

- `--sink-reply` flag:
Addressable sink for the reply events. You can specify the value in the same format/semantic as for `--sink` flag.

- `--sink-dead-letter` flag:
Addressable to serve as dead letter sink. The events which can not be delivered to the configured sink are delivered here. You can specify the value in the same format/semantic as for `--sink` flag.

<hr>

**Examples:**

1 - Create a subscription `sub0` to channel `kafka0` and sink as `event-display-kafka`

```
kn subscription create sub0 --channel kafka0 --sink event-display-kafka
```

2 - Create a subscription with reply sink as broker and dead letter sink as an URL
```
kn subscription create sub1 --channel c0 --sink http://edisplay.default.example.com --sink-reply broker:b0 --sink-dead-letter ksvc:dsl
```

3 - Update sink to an URL for a subscription
```
kn subscription update sub0 --sink http://event-display.default.example.com
```

4 - List subscriptions
```
kn subscription list
NAME   CHANNEL          SUBSCRIBER                                 REPLY       DEAD LETTER SINK   READY   REASON
sub0   Channel:kafka0   http://event-display.default.example.com                                  True    
sub1   Channel:c0       http://edisplay.default.example.com        broker:b0   ksvc:dsl           True 
```

5 - Describe a subscription
```
kn subscription describe sub1
Name:            sub1
Namespace:       default
Annotations:     messaging.knative.dev/creator=minikube-user, messaging.knative.dev/lastModifier=min ...
Age:             43s
Channel:         Channel:c0 (messaging.knative.dev/v1beta1)
Subscriber:      
  URI:           http://edisplay.default.example.com
Reply:           
  Name:          b0
  Resource:      Broker (eventing.knative.dev/v1beta1)
DeadLetterSink:  
  Name:          dsl
  Resource:      Service (serving.knative.dev/v1)

Conditions:  
  OK TYPE                  AGE REASON
  ++ Ready                 43s 
  ++ AddedToChannel        43s 
  ++ ChannelReady          43s 
  ++ ReferencesResolved    43s 
```

6 - Delete a subscription
```
kn subscription delete sub1
```